src/app/modules/payment/selectors.ts

```diff
@@ -0,0 +1,27 @@
+import { AppState as S } from 'store/reducers';
+import { OnChainFeeEstimates } from './types';
+
+export const selectOnChainFees = (s: S) => s.payment.onChainFees;
+export const getAdjustedFees = (s: S): OnChainFeeEstimates | null => {
+  // adjust fees to always be unique since the API may return multiple
+  // speeds with the same number sats/b. This breaks the RadioGroup UI
+  // to have multiple options selected
+  const fees = selectOnChainFees(s);
+  if (!fees) return null;
+
+  const { fastestFee } = fees;
+  let { halfHourFee, hourFee } = fees;
+
+  if (halfHourFee === fastestFee) {
+    halfHourFee--;
+  }
+  if (hourFee >= halfHourFee) {
+    hourFee--;
+  }
```

Comment:
Great catches on this edge case, but another edge case this introduces is if the network is at a very low fee state, the remote possibility that these become `0` could introduce problems. Might be better to _increment_ the other fees instead, for safety's sake.

Also, the `hourFee >= halfHourFee` logic means that if hourFee were 3, and halfHourFee were 2, hourFee would become 2, making them the same. It should probably be `hourFee = halfHourFee - 1`.

Reply:
I was thinking to default to the fastestFee and work backwards, but I didn't consider the case of a `0` fee. Good point. I'll fix it up.

Reply:
I just pushed what I think is a better solution to the original problem. With this selector, I was attempting to fix a UI issue. Multiple RadioButtons were being selected when the fee values from the API were the same for more than one key in the response object. For example: `{ fastestFee: 25, halfHourFee: 25, hourFee: 16 }`. As you saw, my initial solution was to decrement the values slightly to prevent multiple RadioButtons being selected. This in-turn presented other issues with negative numbers like you mentioned. After playing with the adjustments further, I felt like I was adding too much complexity to solve a simple UI problem. I also didn't really like increasing the fee values just to fix the UI. This is someone's money we're spending. Even though it's just a few sats, it didn't sit well in my gut.

I took a step back and looked at the original issue to try to find an alternate solution. I think I came up with something simpler. Instead of using the numeric fees for the RadioButton values (which can contain duplicates), I am now using the response object's key as the value for the RadioButton (ex: `fastestFee` instead of `25`). This solves the multiple button selection UI issue without needing to alter the numeric values. It also eliminates the need for the selector code.

I feel much better about this approach. Let me know what you think.

Reply:
Sounds like a great chop of the Gordian knot. Nice work!